src/Core/Settings.h

```diff
@@ -944,6 +944,7 @@ class IColumn;
     M(Bool, regexp_dict_allow_hyperscan, true, "Allow regexp_tree dictionary using Hyperscan library.", 0) \
     \
     M(Bool, dictionary_use_async_executor, false, "Execute a pipeline for reading from a dictionary with several threads. It's supported only by DIRECT dictionary with CLICKHOUSE source.", 0) \
+    M(Bool, check_sample_dict_key_is_correct, true, "Check primary key type for simple dictionary is Native unsigned integer", 0) \
```

Comment:
I think, `check_dictionary_primary_key` is a better name for the setting because it's actually a primary key.

Reply:
`check_dictionary_primary_key` is better, i will change it.